### PR TITLE
Create data sources for fabric vSphere storage policy and fabric vSph…

### PR DESCRIPTION
**Pull request by zhizhu7**
…ere data stores

Create data sources for fabric vSphere storage policy and data stores for the APIs:
  /iaas/api/fabric-vsphere-datastores
  /iaas/api/fabric-vsphere-storage-policies

Added tests for these two new data sources

### CODE REVIEW

File: vra/data_source_fabric_datastore_vsphere.go [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+	d.SetId(*datastore.ID)
+	d.Set("cloud_account_ids", datastore.CloudAccountIds)
+	d.Set("created_at", datastore.CreatedAt)
+	d.Set("external_id", datastore.ExternalID)
+	d.Set("external_region_id", datastore.ExternalRegionID)
+	d.Set("free_size_gb", datastore.FreeSizeGB)
+	d.Set("name", datastore.Name)
+	d.Set("org_id", datastore.OrgID)
+	d.Set("organization_id", datastore.OrganizationID)
```
Comment (dmettem) [inline vra/data_source_fabric_datastore_vsphere.go:122]: `OrganizationID` is deprecated. You can remove this from the schema.

File: vra/provider.go [hunk 2]
```diff
@@ -45,39 +45,41 @@ func Provider() *schema.Provider {
...
+			"vra_cloud_account_gcp":             dataSourceCloudAccountGCP(),
+			"vra_cloud_account_nsxt":            dataSourceCloudAccountNSXT(),
+			"vra_cloud_account_nsxv":            dataSourceCloudAccountNSXV(),
+			"vra_cloud_account_vmc":             dataSourceCloudAccountVMC(),
+			"vra_cloud_account_vsphere":         dataSourceCloudAccountVsphere(),
+			"vra_data_collector":                dataSourceDataCollector(),
+			"vra_deployment":                    dataSourceDeployment(),
+			"vra_fabric_network":                dataSourceFabricNetwork(),
+			"vra_fabric_datastore_vsphere":      dataSourceFabricDatastoreVsphere(),
+			"vra_fabric_storage_policy_vsphere": dataSourceFabricStoragePolicyVsphere(),
```
Comment (dmettem) [inline vra/provider.go:65]: Rearrange these in the alphabetical order.

File: vra/data_source_fabric_datastore_vsphere_test.go [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccCheckFabricDatastoreVsphereConfig(dsName + "foo"),
+				ExpectError: regexp.MustCompile("fabric vSphere datastore filter doesn't match to any datastore"),
+			},
+			{
+				Config: testAccCheckFabricDatastoreVsphereConfig(dsName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(datasourceName, "name", dsName),
+				),
```
Comment (dmettem) [inline vra/data_source_fabric_datastore_vsphere_test.go:28]: Add the following check as well.
`resource.TestCheckResourceAttrSet(datasourceName, "id")`

File: vra/data_source_fabric_storage_policy_vsphere.go [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+	}
+
+	d.SetId(*storagePolicy.ID)
+	d.Set("cloud_account_ids", storagePolicy.CloudAccountIds)
+	d.Set("created_at", storagePolicy.CreatedAt)
+	d.Set("external_id", storagePolicy.ExternalID)
+	d.Set("external_region_id", storagePolicy.ExternalRegionID)
+	d.Set("name", storagePolicy.Name)
+	d.Set("org_id", storagePolicy.OrgID)
+	d.Set("organization_id", storagePolicy.OrganizationID)
```
Comment (dmettem) [inline vra/data_source_fabric_storage_policy_vsphere.go:113]: Same comment as before.

File: vra/data_source_fabric_storage_policy_vsphere_test.go [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccFabricStoragePolicyVsphereConfig(dsName + "foo"),
+				ExpectError: regexp.MustCompile("fabric vSphere storage policies filter doesn't match to any storage policy"),
+			},
+			{
+				Config: testAccFabricStoragePolicyVsphereConfig(dsName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(datasourceName, "name", dsName),
```
Comment (dmettem) [inline vra/data_source_fabric_storage_policy_vsphere_test.go:27]: Same comment as before.

File: go.sum [hunk 6]
```diff
@@ -278,6 +278,8 @@ github.com/vmihailenco/msgpack v4.0.1+incompatible h1:RMF1enSPeKTlXrXdOcqjFUElyw
 github.com/vmihailenco/msgpack v4.0.1+incompatible/go.mod h1:fy3FlTQTDXWkZ7Bh6AcGMlsjHatGryHQYUTf1ShIgkk=
 github.com/vmware/vra-sdk-go v0.2.11 h1:QSnlKItvEmnhd9yfaCLesTlb/I928F/OhQ7LpaGljyI=
 github.com/vmware/vra-sdk-go v0.2.11/go.mod h1:W6OERth9ssa4FgvS98SOVusGgBNWJs+D4Hq/5ueTgIQ=
+github.com/vmware/vra-sdk-go v0.2.12 h1:kn6EMcIsOxWG0s7JjWWBGihpzP+wGLqvXSSk83wH+TU=
+github.com/vmware/vra-sdk-go v0.2.12/go.mod h1:W6OERth9ssa4FgvS98SOVusGgBNWJs+D4Hq/5ueTgIQ=
```
Comment (dmettem) [inline go.sum:282]: Run `go mod tidy` to clean up the older versions.

**Review by dmettem — CHANGES_REQUESTED** [review]

File: vra/data_source_fabric_datastore_vsphere.go (hunk 1 above)
Comment (zhizhu7) [reply vra/data_source_fabric_datastore_vsphere.go]: Thanks, updated

File: go.sum (hunk 6 above)
Comment (zhizhu7) [reply go.sum:280]: Thanks, cleaned it.

**Review by dmettem — APPROVED** [review]

**Pull request merged**